trace/google/cloud/trace/trace.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Trace for interacting with the Stackdriver Trace API."""
+
+from google.cloud.trace.trace_span import TraceSpan
+from google.cloud.trace.trace_span import format_span_json
+from subprocess import check_output
+
+_GENERATE_TRACE_ID_COMMAND = 'uuidgen | sed s/-//g'
+
+
+class Trace(object):
+    """A trace describes how long it takes for an application to perform
+    an operation. It consists of a set of spans, each of which represent
+    a single timed event within the operation.
```

Comment:
Can you add a note here that traces are *not* thread-safe and must not be shared between threads?